fontforge/tottf.c

```diff
@@ -4499,34 +4499,42 @@ return( NULL );
 return( format12 );
 }
 
+#define SEGMAXINC 50
+#define SEGBREAKEVEN 4
 static FILE *NeedsUCS2Table(SplineFont *sf,int *ucs2len,EncMap *map,int issymbol) {
     /* We always want a format 4 2byte unicode encoding map */
     /* But if it's symbol, only include encodings 0xff20 - 0xffff */
-    uint32 *avail = malloc(65536*sizeof(uint32));
-    int i,j,l;
-    int segcnt, cnt=0, delta, rpos;
-    struct cmapseg { uint16 start, end; uint16 delta; uint16 rangeoff; } *cmapseg;
-    uint16 *ranges;
+    int32 *avail = malloc(65536*sizeof(int32));
```

Comment:
Can you explain the sign change?

Reply:
The previous code stored glyph IDs in `avail` and then used other logic later to go from glyph ID to `ttf_glyph`. The new code stores `ttf_glyph` values in `avail` to start with. Being able to check against -1 is handy but it's mostly to match the type of `ttf_glyph`.